code/src/App.js

```diff
@@ -1,9 +1,91 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
+
+import TaskList from 'components/TaskList';
+import TaskForm from 'components/TaskForm';
 
 export const App = () => {
+  const [taskList, setTaskList] = useState([]);
+  const [loading, setLoading] = useState(false);
+  const [newTodo, setNewTodo] = useState('');
+  // from taskform
+
+  const fetchTasks = () => {
+    setLoading(true);
+    fetch('https://happy-thoughts-technigo.herokuapp.com/thoughts')
+      .then((res) => res.json())
+      .then((data) => setTaskList(data.reverse()))
+      .catch((error) => console.error(error))
+      .finally(() => setLoading(false));
+  }
+
+  // fetchData
+
+  useEffect(() => {
+    fetchTasks();
+  }, []);
+
+  const handleNewTodoChange = (event) => {
+    setNewTodo(event.target.value);
+  }
+  // what daniel calls onNewTaskChange
+
+  const onFormSubmit = (event) => {
+    event.preventDefault();
+
+    const options = {
+      method: 'POST',
+      // because of posting something
+      headers: {
+        'Content-Type': 'application/json'
+      },
+      // what are we sending, contenttype. 'application/json' because we are sending
+      // json in the body (?!?!??!?!??!?!??!?!?!??!)
+      body: JSON.stringify({
+        message: newTodo
+      })
+      // sending what back end wants us to send
+    }
+    fetch('https://happy-thoughts-technigo.herokuapp.com/thoughts', options)
+      .then((res) => res.json())
+      .then(() => fetchTasks())
+      .catch((error) => console.error(error))
+      .finally(() => setNewTodo(''));
+  }
+  // what daniel calls handleFormSubmit
+
+  const onLikesIncrease = () => {
+    const options = {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json'
+      }
+    }
+
+    fetch('https://happy-thoughts-technigo.herokuapp.com/thoughts/_id/like', options)
+      .then((res) => res.json())
+      .then((json) => {
+        console.log(json)
+      })
+      .catch((error) => console.error(error))
+      .finally(() => fetchTasks())
```

Comment:
This function isn't quite ready as I'm sure you know..Inside it the likes (hearts) for a single thought should be increased, something like:
`.then(setSingleThoughtHearts(hearts += 1))`